packages/filesystem/src/node/nsfw-watcher/nsfw-filesystem-watcher.ts

```diff
@@ -111,31 +111,47 @@ export class NsfwFileSystemWatcherServer implements FileSystemWatcherServer {
             this.debug('Files ignored for watching', options.ignored);
         }
 
-        let watcher: nsfw.NSFW | undefined = await nsfw(fs.realpathSync(basePath), (events: nsfw.ChangeEvent[]) => {
-            for (const event of events) {
-                if (event.action === nsfw.actions.CREATED) {
-                    this.pushAdded(watcherId, this.resolvePath(event.directory, event.file!));
-                }
-                if (event.action === nsfw.actions.DELETED) {
-                    this.pushDeleted(watcherId, this.resolvePath(event.directory, event.file!));
-                }
-                if (event.action === nsfw.actions.MODIFIED) {
-                    this.pushUpdated(watcherId, this.resolvePath(event.directory, event.file!));
+        let watcher: nsfw.NSFW | undefined;
+        try {
+            watcher = await nsfw(fs.realpathSync(basePath), (events: nsfw.ChangeEvent[]) => {
+                for (const event of events) {
+                    if (event.action === nsfw.actions.CREATED) {
+                        this.pushAdded(watcherId, this.resolvePath(event.directory, event.file!));
+                    }
+                    if (event.action === nsfw.actions.DELETED) {
+                        this.pushDeleted(watcherId, this.resolvePath(event.directory, event.file!));
+                    }
+                    if (event.action === nsfw.actions.MODIFIED) {
+                        this.pushUpdated(watcherId, this.resolvePath(event.directory, event.file!));
+                    }
+                    if (event.action === nsfw.actions.RENAMED) {
+                        this.pushDeleted(watcherId, this.resolvePath(event.directory, event.oldFile!));
+                        this.pushAdded(watcherId, this.resolvePath(event.newDirectory || event.directory, event.newFile!));
+                    }
                 }
-                if (event.action === nsfw.actions.RENAMED) {
-                    this.pushDeleted(watcherId, this.resolvePath(event.directory, event.oldFile!));
-                    this.pushAdded(watcherId, this.resolvePath(event.newDirectory || event.directory, event.newFile!));
+            }, {
+                errorCallback: error => {
+                    // see https://github.com/atom/github/issues/342
+                    console.warn(`Failed to watch "${basePath}":`, error);
+                    if (error === 'Inotify limit reached') {
+                        if (this.client) {
+                            this.client.onError();
+                        }
+                    }
+                    this.unwatchFileChanges(watcherId);
+                },
+                ...this.options.nsfwOptions
+            });
+            await watcher.start();
+        } catch (error) {
+            if (error.message === 'NSFW was unable to start watching that directory.') {
```

Comment:
Do you know how to land in this state?

Reply:
This is the error I was getting back when I first worked on this.  After rebasing etc I started getting the Inotify message in the callback.  It may have been a bad build.  I was never happy with this code here, and the message is too generic so may be caused by other problems, perhaps on Windows and Mac.  I will spend more time seeing if I hit this code again and, if not, I'll remove it.

Reply:
I've now removed this check as I cannot reproduce the condition that caused it.  I also removed the link to instructions in the browser version as the user can't change the inotify limit on the server.